### PR TITLE
chore: close HUDs with ESC key

### PR DESCRIPTION
**Pull request by AjimenezDCL**
#1379 

Some HUDs didnt have the ESC key implemented.  I took the opportunity to achieve some mini-refactor with some HUDS implementing the MVC wrongly.

### CODE REVIEW

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public void Initialize(UserProfile userProfile, WearableDictionary catalog, bool
         view = AvatarEditorHUDView.Create(this);
 
         view.OnToggleActionTriggered += ToggleVisibility;
-        view.OnCloseActionTriggered += Hide;
+        view. OnCloseActionTriggered += Hide;
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs]: I think that you put a space by accident.

**Review by sandrade-dcl — CHANGES_REQUESTED** [review]
I found a issue with this UI that I think we didn't take into account:
- **Email Prompt HUD**: The window is not being closed pressing `ESC`.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs (hunk 1 above)
Comment (eordano) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs]:
```suggestion
        view.OnCloseActionTriggered += Hide;
```

**Review by pravusjif — CHANGES_REQUESTED** [review]
* Every UI I close with ESC then can't be opened again with its hotkey (for example X for Explore, TAB/M for Navmap) until i lock the mouse back. If we don't use the ESC key, and close the HUDs normally, we can re-open them with no problem with the hotkey even though the mouse isn't locked.

* I still couldn't close the email prompt with the ESC key, for easier testing, if you enter as guest you can see that HUD appear after a couple of minutes later

**Comment by sandrade-dcl** [comment]
> * Every UI I close with ESC then can't be opened again with its hotkey (for example X for Explore, TAB/M for Navmap) until i lock the mouse back. If we don't use the ESC key, and close the HUDs normally, we can re-open them with no problem with the hotkey even though the mouse isn't locked.

I also confirm that this is happening.

> * I still couldn't close the email prompt with the ESC key, for easier testing, if you enter as guest you can see that HUD appear after a couple of minutes later

@pravusjif I have just test it and it worked for me... Maybe you executed some cached version?

**Comment by AjimenezDCL** [comment]
@pravusjif Not being able to use the hotkeys after pressing ESC is not actually an issue from this PR, you can experiment the same behaviour in production (and it's fixed by regaining focus in the game). 

I would suggest opening a new bug to solve that.

**Review by pravusjif — APPROVED** [review]

**Review by sandrade-dcl — APPROVED** [review]

**Pull request merged**